File: hash_insertion_unit.test.js
describe('HashMap tests', () => {
    
  // Test to add a simple element
  test('should add an element to the HashMap', () => {
      let hashmap = new Map();
      hashmap.set('key1', 'value1');
      expect(hashmap.get('key1')).toBe('value1');
  });

  // Edge case: Overwriting an existing key
  test('should overwrite an existing key', () => {
      let hashmap = new Map();
      hashmap.set('key1', 'value1');
      hashmap.set('key1', 'newValue');
      expect(hashmap.get('key1')).toBe('newValue');
  });

  // Edge case: Handling a large number of elements
  test('should handle large number of elements', () => {
      let hashmap = new Map();
      for (let i = 0; i < 10000; i++) {
          hashmap.set(`key${i}`, `value${i}`);
      }
      expect(hashmap.size).toBe(10000);
  });

});

File: hashInsertionUnitTest.java
import static org.junit.Assert.*;
import org.junit.Test;
import java.util.HashMap;

public class HashMapTest {

    // Test to add a simple element
    @Test
    public void testAddElement() {
        HashMap<String, String> hashmap = new HashMap<>();
        hashmap.put("key1", "value1");
        assertEquals("value1", hashmap.get("key1"));
    }

    // Edge case: Overwriting an existing key
    @Test
    public void testOverwriteElement() {
        HashMap<String, String> hashmap = new HashMap<>();
        hashmap.put("key1", "value1");
        hashmap.put("key1", "newValue");
        assertEquals("newValue", hashmap.get("key1"));
    }

    // Edge case: Adding a large number of elements (for performance)
    @Test
    public void testLargeNumberOfElements() {
        HashMap<String, String> hashmap = new HashMap<>();
        for (int i = 0; i < 10000; i++) {
            hashmap.put("key" + i, "value" + i);
        }
        assertEquals("The map should contain 10000 elements", 10000, hashmap.size());
    }
}
